src/dodal/devices/eiger.py

```diff
@@ -98,74 +96,109 @@ def unstage(self) -> bool:
         self.odin.file_writer.start_timeout.put(1)
         LOGGER.info("Waiting on filewriter to finish")
         self.filewriters_finished.wait(30)
+
         LOGGER.info("Disarming detector")
         self.disarm_detector()
+
         status_ok = self.odin.check_odin_state()
         self.disable_roi_mode()
         return status_ok
 
+    # this is no longer used TODO clean up
     def enable_roi_mode(self):
         self.change_roi_mode(True)
 
     def disable_roi_mode(self):
         self.change_roi_mode(False)
 
-    def change_roi_mode(self, enable: bool):
+    def change_roi_mode(self, enable: bool) -> Status:
         assert self.detector_params is not None
         detector_dimensions = (
             self.detector_params.detector_size_constants.roi_size_pixels
             if enable
             else self.detector_params.detector_size_constants.det_size_pixels
         )
 
-        status = self.cam.roi_mode.set(1 if enable else 0)
-        status &= self.odin.file_writer.image_height.set(detector_dimensions.height)
-        status &= self.odin.file_writer.image_width.set(detector_dimensions.width)
-        status &= self.odin.file_writer.num_row_chunks.set(detector_dimensions.height)
-        status &= self.odin.file_writer.num_col_chunks.set(detector_dimensions.width)
-
-        status.wait(10)
+        status = self.cam.roi_mode.set(
+            1 if enable else 0, timeout=self.GENERAL_STATUS_TIMEOUT
+        )
+        status &= self.odin.file_writer.image_height.set(
+            detector_dimensions.height, timeout=self.GENERAL_STATUS_TIMEOUT
+        )
+        status &= self.odin.file_writer.image_width.set(
+            detector_dimensions.width, timeout=self.GENERAL_STATUS_TIMEOUT
+        )
+        status &= self.odin.file_writer.num_row_chunks.set(
+            detector_dimensions.height, timeout=self.GENERAL_STATUS_TIMEOUT
+        )
+        status &= self.odin.file_writer.num_col_chunks.set(
+            detector_dimensions.width, timeout=self.GENERAL_STATUS_TIMEOUT
+        )
```

Comment:
Could: You're right that these should all have a timeout but it's getting messy with them set so many times here. What about something like:
```python
status = Status(timeout=self.GENERAL_STATUS_TIMEOUT) # Overall status to track timeout for the whole operation

status &= self.cam.roi_mode.set(1 if enable else 0)
...
```

Reply:
And we could repeat the pattern elsewhere

Reply:
I think if we did this then the first part of the status would never automatically be set as finished. Would attaching a timeout to just one of these statuses give the same effect?

Reply:
As discussed we'll leave this as it is for now